pkg/util/version_history.go

```diff
@@ -14,11 +14,19 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/version"
+	"gopkg.in/yaml.v2"
 )
 
 type versionHistoryEntry struct {
-	Version   string    `json:"version"`
-	Timestamp time.Time `json:"timestamp"`
+	Version       string        `json:"version"`
+	Timestamp     time.Time     `json:"timestamp"`
+	InstallMethod installMethod `json:"install_method" yaml:"install_method"`
+}
+
+type installMethod struct {
+	Tool             string `json:"tool" yaml:"tool"`
+	ToolVersion      string `json:"tool_version" yaml:"tool_version"`
+	InstallerVersion string `json:"installer_version" yaml:"installer_version"`
```

Comment:
I wanted to use the following type, but it causes cycle import.
https://github.com/DataDog/datadog-agent/blob/af25c37f661b19087f0d791d5be534c5924af3f0/pkg/metadata/host/payload.go#L51-L56

Reply:
The alternative would be to move the definition somewhere else that both places can import, but I think I prefer the current solution better.